src/app-layer-htp-file.c

```diff
@@ -225,34 +228,108 @@ int HTPParseContentRange(bstr * rawvalue, HtpContentRange *range)
  *  \retval -2 error parsing
  *  \retval -3 error negative end in range
  */
-int HTPFileSetRange(HtpState *s, bstr *rawvalue)
+int HTPFileOpenWithRange(HtpState *s, const uint8_t *filename, uint16_t filename_len,
+        const uint8_t *data, uint32_t data_len, uint64_t txid, bstr *rawvalue, HtpTxUserData *htud)
 {
     SCEnter();
+    uint16_t flags;
 
     if (s == NULL) {
         SCReturnInt(-1);
     }
 
+    HtpContentRange crparsed;
```

Comment:
should this whole logic be wrapped in a `if (STREAM_TOCLIENT) {`? It seems all calls to `HTPFileOpen` assume toclient

Reply:
`HTPFileOpenWithRange` is only called from `HtpResponseBodyHandle` so only `STREAM_TOCLIENT